lib/grpmat.gi

```diff
@@ -1241,6 +1241,146 @@ InstallGlobalFunction( "BlowUpIsomorphism", function( matgrp, B )
     end );
 
 
+#############################################################################
+##
+#F  DiagonalAutomorphismGroupOfSL(<G>)
+## 
+##  Returns the group of all diagonal automorphsims of G = SL(d,q)
+##  An automorphism f : G -> G is a diagonal automorphism iff
+##  f(U) = D U D^-1 for all U in G and some diagonal matrix D in GL(d,q)
+##
+InstallGlobalFunction("DiagonalAutomorphismGroupOfSL",function(G)
+    local D,x,z,i,M,d,q;
+    d := SLDegree(G);
+    q := Size(SLUnderlyingField(G));
```

Comment:
Don't use `SLDegree` and `SLUnderlyingField`, they are not for this kind of application. Instead you probably will want to check whether `IsNaturalSL(g)` gives `true` at the start, and then just use `DegreeOfMatrixGroup` and `FieldOfMatrixGroup`